### PR TITLE
Fix log links in K8s Array tasks

### PR DESCRIPTION
**Pull request by anandswaminathan**
Currently Log links are not sent to Admin/UI for K8s batch array task due to a bug. Similarly there were other cases that were breaking.

* Log links for remote cluster execution.
* Log links for finished tasks
* Introduces finalizer for pods created by K8s array task.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

PTAL @EngHabu @katrogan

### CODE REVIEW

File: go/tasks/logs/logging_utils.go [hunk 1]
```diff
@@ -9,13 +9,15 @@ import (
...
 
-func GetLogsForContainerInPod(ctx context.Context, pod *v1.Pod, index uint32, nameSuffix string) ([]*core.TaskLog, error) {
+func GetLogsForContainerInPodCustomURL(ctx context.Context, pod *v1.Pod, index uint32, nameSuffix, kubernetesURL string) ([]*core.TaskLog, error) {
 	var logs []*core.TaskLog
 	logConfig := GetLogConfig()
 
 	logPlugins := map[string]logUtils.LogPlugin{}
 
-	if logConfig.IsKubernetesEnabled {
+	if kubernetesURL != "" {
```
Comment (EngHabu) [inline go/tasks/logs/logging_utils.go:18]:
```suggestion
	if len(kubernetesURL) > 0 {
```

File: go/tasks/plugins/array/k8s/task.go [hunk 2]
```diff
@@ -82,6 +104,7 @@ func (t Task) Launch(ctx context.Context, tCtx core.TaskExecutionContext, kubeCl
 	pod = ApplyPodPolicies(ctx, t.Config, pod)
 	pod = applyNodeSelectorLabels(ctx, t.Config, pod)
 	pod = applyPodTolerations(ctx, t.Config, pod)
+	pod = addPodFinalizer(pod)
```
Comment (EngHabu) [inline go/tasks/plugins/array/k8s/task.go:107]: We should gate this on the same or similar config: https://github.com/lyft/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/config/config.go#L53

File: go/tasks/plugins/array/k8s/task.go [hunk 3]
```diff
@@ -180,8 +234,26 @@ func (t Task) Finalize(ctx context.Context, tCtx core.TaskExecutionContext, kube
...
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			logger.Errorf(ctx, "Error fetching pod [%s] in Finalize [%s]", podName, err)
+			return err
+		}
+	} else {
+		pod = clearFinalizer(pod)
+		err := kubeClient.GetClient().Update(ctx, pod)
+		if err != nil {
+			logger.Errorf(ctx, "Error updating pod finalizer [%s] in Finalize [%s]", podName, err)
```
Comment (EngHabu) [inline go/tasks/plugins/array/k8s/task.go:251]: We should check for notFound here again... and not error if it's

**Comment by katrogan** [comment]
LGTM

File: go/tasks/plugins/array/k8s/task.go (hunk 3 above)
Comment (anandswaminathan) [reply go/tasks/plugins/array/k8s/task.go:251]: Why? If pod is found, how can it disappear before update ? I guess no harm in adding.

**Pull request closed without merging**